src/components/TodoItem.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+
+type Props = {
+  todo: Todo;
+  onDelete: (id: number) => void;
+  isLoading: boolean;
+  onToggle: (todo: Todo) => void;
+  onUpdate: (updatedTodo: Todo) => Promise<void>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  isLoading,
+  onToggle,
+  onUpdate,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const editFieldRef = useRef<HTMLInputElement>(null);
+
+  const handleSubmit = () => {
+    if (!newTitle.trim()) {
+      onDelete(todo.id);
+    } else {
+      onUpdate({
+        ...todo,
+        title: newTitle.trim(),
+      });
+    }
+
+    setIsEditing(false);
+  };
+
+  const handlePressedKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      handleSubmit();
+    }
+
+    if (event.key === 'Escape') {
+      setIsEditing(false);
+    }
+  };
+
+  useEffect(() => {
+    if (isEditing && editFieldRef.current) {
+      editFieldRef.current.focus();
+    }
+  }, [isEditing]);
+
+  return (
+    <div
+      key={todo.id}
+      data-cy="Todo"
+      className={classNames('todo', { completed: todo.completed })}
+    >
+      {/* eslint-disable jsx-a11y/label-has-associated-control */}
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
```

Comment:
Consider using destruction assignment for `todo` object

Reply:
Not fixed